generator/src/main/java/com/graphicsfuzz/generator/tool/Generate.java

```diff
@@ -202,7 +213,42 @@ public static StringBuilder generateVariant(ShaderJob shaderJob,
     }
 
     if (args.getGenerateUniformBindings()) {
-      shaderJob.makeUniformBindings();
+      boolean supportedInAllShaders = true;
+      for (TranslationUnit tu : shaders) {
+        if (!tu.getShadingLanguageVersion().supportedPushConstants()) {
+          supportedInAllShaders = false;
+        }
+      }
+      if (supportedInAllShaders) {
+        if (random.nextFloat() <= args.getPushConstantProbability()) {
+          // Get the list of (unique) uniform names in all shaders
```

Comment:
add to comment the fact that we restrict to uniforms that can definitely be made into push constants.

Reply:
done